src/databudgie/adapter/postgres.py

```diff
@@ -75,11 +75,13 @@ def export_table_ddl(self, table_name: str, console: Console = default_console):
 
         return pg_dump(url, f"--schema-only -t {table_name}")
 
-    def reset_database(self):
+    def reset_database(self) -> None:
         """Attempt to kill the existing database and bring it back up."""
         connection = self.session.connection()
+
+        assert connection.engine.url
         url: URL = connection.engine.url
-        database = url.database
+        database = cast(str, url.database)
```

Comment:
what is the significance of all these casts

Reply:
mypy complains because they're typed as optional strings, but in practical use, they're always values. I could alternatively probably `assert url.database`